predicators/nsrt_learning/strips_learning/base_strips_learner.py

```diff
@@ -43,14 +43,43 @@ def learn(self) -> List[PartialNSRTAndDatastore]:
         filtering may break it.
         """
         learned_pnads = self._learn()
+        # Remove pnads by increasing min_data_perc until harmlessness breaks.
+        if CFG.enable_harmless_op_pruning:
+            assert self._verify_harmlessness
+            assert not CFG.disable_harmlessness_check
+        # Keeps track of latest set of harmless pnads. Will interating over
+        # all min_perc_data_for_nsrts until it breaks.
+        min_harmless_pnads = learned_pnads
+        for min_perc_data_for_nsrt in range(CFG.min_perc_data_for_nsrt, 100):
+            learned_pnads = self._learn()
+            min_data = max(CFG.min_data_for_nsrt,
+                           self._num_segments * min_perc_data_for_nsrt / 100)
+            learned_pnads = [
+                pnad for pnad in learned_pnads
+                if len(pnad.datastore) >= min_data
+            ]
+            if self._verify_harmlessness and \
+                not CFG.disable_harmlessness_check:
+                logging.info("\nRunning harmlessness check...")
+                if not self._check_harmlessness(learned_pnads):
+                    break
+            else:
+                # If we are not verifying harmlessness than we assume that
```

Comment:
the logic here is a bit confusing to follow for the cases we had before (where harmless op pruning is not enabled). could we refactor this a bit so that all that happens in this for loop is we determine a good `min_perc_data_for_nsrt` to use, and the for loop only gets run in the case where `enable_harmless_op_pruning` is True; and then outside of the for loop, we use the determined `min_perc_data_for_nsrt` to construct the final PNADs?